tests/ActivityApiTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+use Rogue\Models\Signup;
+use Rogue\Models\Event;
+use Faker\Generator;
+
+class ActivityApiTest extends TestCase
+{
+    /*
+     * Base URL for the Api.
+     */
+    protected $activityApiUrl = 'api/v2/activity';
+
+    /**
+     * Test for retrieving a user's activity with limit query param.
+     *
+     * GET /activity?limit=8
+     * @return void
+     */
+    public function testActivityIndexWithLimitQuery()
+    {
+        $this->json('GET', $this->activityApiUrl . '?limit=8');
+
+        $this->assertResponseStatus(200);
+
+        $this->seeJsonSubset([
+            'meta' => [
+                'pagination' => [
+                    'per_page' => 8
+                ],
+            ]
+        ]);
+    }
+
+    /**
+     * Test for retrieving a user's activity with page query param.
+     *
+     * GET /activity?page=3
+     * @return void
+     */
+    public function testActivityIndexWithPageQuery()
+    {
+        $this->json('GET', $this->activityApiUrl . '?page=3');
+
+        $this->assertResponseStatus(200);
+
+        $this->seeJsonSubset([
+            'meta' => [
+                'pagination' => [
+                    'current_page' => 3
+                ],
+            ]
+        ]);
+    }
+
+    /**
+     * Test for retrieving a user's activity with campaign_id query param.
+     *
+     * GET /activity?filter[campaign_id]=47
+     * @return void
+     */
+    public function testActivityIndexWithCampaignIdQuery()
+    {
+        $signup = factory(Signup::class)->create();
+        $event = Event::all()->first();
+        $event->signup_id = $signup->id;
+        $event->save();
+
```

Comment:
Yeah, `Event::all()->first()` definitely seems a little dangerous. I'm still a little confused how the relationship between these two records works... it seems weird that both tables need a `RELATED_id` column to mark the relation.

Let's go with what you have here for now, but make a ticket to revisit later?